session/src/main/java/org/apache/iotdb/session/SessionConnection.java

```diff
@@ -722,20 +751,27 @@ protected void testInsertTablets(TSInsertTabletsReq request)
   private boolean reconnect() {
     boolean flag = false;
     for (int i = 1; i <= Config.RETRY_NUM; i++) {
```

Comment:
Always try the first RETRY_NUM clients, it's not good. Maybe you can record a nextHostIndex to try next hostId.

Reply:
in new commit,Instead of trying the first node in the list every time, try the current node again, and then try the next node in the list

Reply:
ok, in new commit,connect nodes randomly